### ISSUE TITLE
trouble wit API

### ISSUE THREAD

**Issue by CGEdU** [stackblitz-labs/bolt.diy#287]
### Describe the bug

I'm having trouble with Bolt and the OpenAI API. I followed the installation steps and managed to run Bolt locally, but when I send the prompt, I get an API error. I don’t understand why this happens or if I’m doing something incorrectly. Can someone help me? I’ve already deleted that API key, haha—I just left it here as an example of how I’m using it, in case there's something wrong that needs fixing. Thanks!

### Link to the Bolt URL that caused the error

https://github.com/coleam00

### Steps to reproduce

I'm having trouble with Bolt and the OpenAI API. I followed the installation steps and managed to run Bolt locally, but when I send the prompt, I get an API error. I don’t understand why this happens or if I’m doing something incorrectly. Can someone help me? I’ve already deleted that API key, haha—I just left it here as an example of how I’m using it, in case there's something wrong that needs fixing. Thanks!

### Expected behavior

I need help :C 

### Screen Recording / Screenshot

Mod Edit: Removing screenshot including API key

### Platform

Windows

**Comment by chrismahoney** [comment]
Hey there, I removed the screenshot with sensitive info, I would suggest that you recreate the key that you accidentally posted here. Just saw that you mentioned it, but just being proactive to prevent any malicious use.

Can you provide some additional details about the specific error you encountered? Thanks

**Comment by przbadu** [comment]
This is the correct link:

https://github.com/coleam00/bolt.new-any-llm.git

**Comment by Prathamesh-Shanbhag** [comment]
Yep facing the same issue - Looks like someone has merged bad code in maybe? The whole thing no longer works with OpenAI API keys

**Comment by Mrtlearns** [comment]
I have the same issue. I am stating the the docker with docker-compose --profile production up and maintained the .env and .env.local with exactly the same [7] different API keys (Groq, openai, etc etc)
None of them work and neither does using the pencil and entering it manually.  It is almost like it is not reading the .env or even the manual input.

**Comment by RiseWith** [comment]
> Yep facing the same issue - Looks like someone has merged bad code in maybe? The whole thing no longer works with OpenAI API keys

Any idea which one we should roll-back to?

**Comment by amehat** [comment]
same issue

**Comment by chrismahoney** [comment]
I think this is related to a few other bugs, goal is to have .env.local 100% work with docker compose - will be taking a look along with #306.

**Comment by RiseWith** [comment]
> I think this is related to a few other bugs, goal is to have .env.local 100% work with docker compose - will be taking a look along with #306.

That would be great. This is an exciting tool and it's a shame that Docker has issues. Good luck

**Comment by RiseWith** [comment]
Is there another way that I can use the repo? Because it seems like the app won’t work without this bug being fixed

**Comment by dustinwloring1988** [comment]
please try the latest version if you are still having problems let me know

**Comment by aliasfoxkde** [comment]
I'm guessing that you need to add payment processing to use the OpenAI API key with it. If you want to test it out without needing to setup, or deploy, please use: https://ottodev.cyopsys.com/

You'll need to provide your own keys, but they are secure in your local storage, and it automatically updates with the latest repo changes.

Details: https://thinktank.ottomator.ai/t/no-setup-required-batteries-not-included/1852

**Comment by Bitboy71960** [comment]
unresolved problem, does anyone still have a version of bolt that works with openai ?

**Comment by thecodacus** [comment]
can you try with the latest stable branch. or main. and set the api key on the UI.
should be working.

**Comment by KristofNagyArch** [comment]
Same issue.

![Screenshot 2025-01-07 at 22 09 41](https://github.com/user-attachments/assets/6073479d-63b5-4269-ace1-0d7f0220dac1)
![Screenshot 2025-01-07 at 22 10 05](https://github.com/user-attachments/assets/90aedfd4-f08c-4b14-acbd-08a00a95508b)
![Screenshot 2025-01-07 at 22 10 17](https://github.com/user-attachments/assets/2fb27860-0689-418b-b771-3b65d010f68e)

**Comment by ClouDevops101** [comment]
Still same issue, even when working with the stable branch,, Deepseek [KO], openAI  [KO], only working with google Gemini[OK],  any Fix ?

**Comment by leex279** [comment]
As @thecodacus mentioned please try the latest "main" branch and take a look at your terminal where you startet bolt.diy, as the there is improved logging for the terminal, which should tell you, what the actual problem is.

**Comment by leex279** [comment]
Closed for know: If still a problem, open a topic in community please to investigate deeper: https://thinktank.ottomator.ai/c/bolt-diy/bolt-diy-issues-and-troubleshooting/22
